lightning/src/ln/channelmanager.rs

```diff
@@ -1451,20 +1451,30 @@ macro_rules! remove_channel {
 
 macro_rules! send_channel_ready {
 	($self: ident, $pending_msg_events: expr, $channel: expr, $channel_ready_msg: expr) => {{
-		$pending_msg_events.push(events::MessageSendEvent::SendChannelReady {
-			node_id: $channel.get_counterparty_node_id(),
-			msg: $channel_ready_msg,
-		});
-		// Note that we may send a `channel_ready` multiple times for a channel if we reconnect, so
-		// we allow collisions, but we shouldn't ever be updating the channel ID pointed to.
-		let mut short_to_chan_info = $self.short_to_chan_info.write().unwrap();
-		let outbound_alias_insert = short_to_chan_info.insert($channel.outbound_scid_alias(), ($channel.get_counterparty_node_id(), $channel.channel_id()));
-		assert!(outbound_alias_insert.is_none() || outbound_alias_insert.unwrap() == ($channel.get_counterparty_node_id(), $channel.channel_id()),
-			"SCIDs should never collide - ensure you weren't behind the chain tip by a full month when creating channels");
-		if let Some(real_scid) = $channel.get_short_channel_id() {
-			let scid_insert = short_to_chan_info.insert(real_scid, ($channel.get_counterparty_node_id(), $channel.channel_id()));
-			assert!(scid_insert.is_none() || scid_insert.unwrap() == ($channel.get_counterparty_node_id(), $channel.channel_id()),
+		if $self.default_configuration.manually_signal_channel_ready {
```

Comment:
well at least for our use case we only care about inbound channels.  I'm not sure what _other_ use cases there might be for this feature other than waiting to see the funding transaction used by your counterparty.

feels like any potential use case for outbound channels could be implemented during the handling of FundingGenerationReady and delaying giving the tx back to ldk then.

I guess in theory(?) there's some use case where you want to do some check _after_ receiving funding_signed.  I don't know, happy to put the config in either place.

I think it doesn't really change the implementation details either way afaict since we will need (at least one) flag in Channel to monitor readiness anyway.